fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/service/LoanScheduleAssembler.java

```diff
@@ -466,7 +468,9 @@ private LoanApplicationTerms assembleLoanApplicationTermsFrom(final JsonElement
         final boolean isPrincipalCompoundingDisabledForOverdueLoans = this.configurationDomainService
                 .isPrincipalCompoundingDisabledForOverdueLoans();
 
-        final boolean isDownPaymentEnabled = loanProduct.getLoanProductRelatedDetail().isEnableDownPayment();
+        final boolean isDownPaymentEnabled = loanScheduleValidator
```

Comment:
Validations are called when loan got submitted or updated:
`LoanApplicationWritePlatformServiceJpaRepositoryImpl.submitApplication`
and
`LoanApplicationWritePlatformServiceJpaRepositoryImpl.modifyApplication`

Please move the validations into the `loanApplicationValidator.validateForCreate` and `loanApplicationValidator.validateForModify`